common/tier4_automatic_goal_rviz_plugin/src/automatic_goal_panel.cpp

```diff
@@ -0,0 +1,484 @@
+    std::bind(&AutowareAutomaticGoalPanel::onAppendGoal, this, std::placeholders::_1));
+  initCommunication(raw_node_.get());
+}
+
+void AutowareAutomaticGoalPanel::onToggleLoopList(bool checked)
+{
+  is_loop_list_on = checked;
+  updateGUI();
+}
+
+void AutowareAutomaticGoalPanel::onToggleSaveGoalsAchievement(bool checked)
+{
+  if (checked) {
+    QString file_name = QFileDialog::getSaveFileName(
+      this, tr("Save File with  GoalsList"), "/home/goals_achieved.log",
+      tr("Achieved goals (*.log)"));
+    goals_achiev_file_path_ = file_name.toStdString();
+  } else {
+    goals_achiev_file_path_ = "";
+  }
+  updateGUI();
+}
+
+void AutowareAutomaticGoalPanel::onToggleAutoMode(bool checked)
+{
+  if (checked && goals_list_widget_ptr_->selectedItems().count() != 1) {
+    showMessageBox("Select the first goal in GoalsList");
+    automatic_mode_btn_ptr_->setChecked(false);
+  } else {
+    if (checked) current_goal = goals_list_widget_ptr_->currentRow();
+    is_automatic_mode_on = checked;
+    is_automatic_mode_on ? qtimer_->start(1000) : qtimer_->stop();
+    onClickClearRoute();  // here will be set State::AUTONEXT or State::EDITING;
+  }
+}
+
+void AutowareAutomaticGoalPanel::onClickPlan()
+{
+  if (goals_list_widget_ptr_->selectedItems().count() != 1) {
+    showMessageBox("Select a goal in GoalsList");
+    return;
+  }
+
+  if (callPlanToGoalIndex(cli_set_route_, goals_list_widget_ptr_->currentRow())) {
+    state = State::PLANNING;
+    updateGUI();
+  }
+}
+
+void AutowareAutomaticGoalPanel::onClickStart()
+{
+  if (callService<ChangeOperationMode>(cli_change_to_autonomous_)) {
+    state = State::STARTING;
+    updateGUI();
+  }
+}
+
+void AutowareAutomaticGoalPanel::onClickStop()
+{
+  // if ERROR is set then the ego is already stopped
+  if (state == State::ERROR) {
+    state = State::STOPPED;
+    updateGUI();
+  } else if (callService<ChangeOperationMode>(cli_change_to_stop_)) {
+    state = State::STOPPING;
+    updateGUI();
+  }
+}
+
+void AutowareAutomaticGoalPanel::onClickClearRoute()
+{
+  if (callService<ClearRoute>(cli_clear_route_)) {
+    state = State::CLEARING;
+    updateGUI();
+  }
+}
+
+void AutowareAutomaticGoalPanel::onClickRemove()
+{
+  if (static_cast<unsigned>(goals_list_widget_ptr_->currentRow()) < goals_list_.size())
+    goals_list_.erase(goals_list_.begin() + goals_list_widget_ptr_->currentRow());
+  resetAchievedGoals();
+  updateGUI();
+  updateGoalsList();
+}
+
+void AutowareAutomaticGoalPanel::onClickLoadListFromFile()
+{
+  QString file_name = QFileDialog::getOpenFileName(
+    this, tr("Open File with GoalsList"), "/home", tr("Goal lists (*.yaml)"));
+  if (file_name.count() > 0) loadGoalsList(file_name.toStdString());
+}
+
+void AutowareAutomaticGoalPanel::onClickSaveListToFile()
+{
+  if (goals_list_.size() > 0) {
+    QString file_name = QFileDialog::getSaveFileName(
+      this, tr("Save File with  GoalsList"), "/home/goals_list.yaml", tr("Goal lists (*.yaml)"));
```

Comment:
`/home/goals_list.yaml` I think this kind of file path should be `log/` or `/tmp/hoge`

Reply:
similar discussion
https://github.com/autowarefoundation/autoware/pull/3031